tests/python_package_test/test_dask.py

```diff
@@ -1229,12 +1222,9 @@ def test_training_succeeds_when_data_is_dataframe_and_label_is_column_array(
     client.close(timeout=CLIENT_CLOSE_TIMEOUT)
 
 
-@pytest.mark.parametrize('task', tasks)
+@pytest.mark.parametrize('task', ['binary-classification', 'ranking', 'regression'])
```

Comment:
> I noticed that test_init_score fails for the multiclass-classification task but I removed it from the tasks in that test and can make a different PR adding it back once #4046 is solved.

This PR should include a test for `init_score` used for multiclass classification. I agree with you that the "pass a 1D array" interface could lead to mistakes in the multiclass classification case, but I don't consider #4046 a bug and I expect that a PR that addresses #4046 would only ADD the ability to pass an array of shape `(n_samples, n_classes)`, not remove the current behavior of passing a 1D array. I expect that it will end up that way so that we don't break existing users' code.

Here's a minimal example showing that the current 1D-array behavior can work in the Dask interface.

```python
from sklearn.datasets import make_blobs
import lightgbm as lgb
import numpy as np
import dask.array as da
from distributed import Client, LocalCluster

cluster = LocalCluster(n_workers=2)
client = Client(cluster)

X, y = make_blobs(n_samples=1000, n_features=50, centers=3)

dX = da.from_array(X, chunks=(100, 50))
dy = da.from_array(y, chunks=(100,))

init_scores = dy.map_blocks(
    lambda x: np.repeat(0.8, x.size * 3),
    dtype=np.float64
)

assert init_scores.npartitions == dy.npartitions

dask_model = lgb.DaskLGBMClassifier()
dask_model.fit(dX, dy, init_score=init_scores)

assert dask_model.booster_.trees_to_dataframe()['value'][0] == 0.0
```

So for this test, can you please just change the `init_score` setup to something like this?

```python
# init_scores must be a 1D array, even for multiclass classification
# where you need to provide 1 score per class for each row in X
# https://github.com/microsoft/LightGBM/issues/4046
size_factor = 1
if task == "multiclass-classification":
    size_factor = 3

if output.startswith('dataframe'):
    init_scores = dy.map_partitions(
        lambda x: pd_Series(np.repeat(init_score, x.size * size_factor))
    )
else:
    init_scores = dy.map_blocks(
        lambda x: np.repeat(init_score, x.size * size_factor),
        dtype=np.float64
    )
```